### PR TITLE
WIP Feature/3263 preview start using symbols in preview

### PR DESCRIPTION
**Pull request by codehag**
Associated Issue: #3263

### CODE REVIEW

File: src/utils/parser/getSymbols.js [hunk 1]
```diff
@@ -119,6 +119,16 @@ function extractSymbols(source: Source) {
         });
       }
 
+      if (t.isThisExpression(path.node)) {
+        const { start, end } = path.node.loc;
+        memberExpressions.push({
```
Comment (jasonLaster) [inline src/utils/parser/getSymbols.js]: this is not a memberExpressions, maybe it should be an identifier?

Comment (jasonLaster) [inline src/utils/parser/getSymbols.js]: we should add a test to `fixture/expressions`

File: src/actions/ast.js [hunk 2]
```diff
@@ -79,6 +79,20 @@ export function clearSelection() {
   };
 }
 
+function findSymbol(symbols, tokenPos) {
+  const array = Object.keys(symbols).reduce(
```
Comment (jasonLaster) [inline src/actions/ast.js]: we probably need an explicit order here to prefer more specific types like `member expression`...

Or not if the start/end loc help out

File: src/utils/parser/getSymbols.js (hunk 1 above)
Comment (codehag) [inline src/utils/parser/getSymbols.js]: :+1:

**Pull request merged**